src/lib/coda_metrics/coda_metrics.ml

```diff
@@ -90,6 +90,34 @@ module TextFormat_0_0_4 = struct
           samples )
 end
 
+module Moving_average (Spec : sig
+  val tick_interval : Core.Time.Span.t
+
+  val rolling_interval : Core.Time.Span.t
+
+  val display : Float.t -> Float.t
+end) =
+struct
+  open Async
+
+  let total = ref 0.
+
+  let create () ~name ~subsystem ~namespace ~help =
+    let gauge = Gauge.v name ~subsystem ~namespace ~help in
+    let rec tick () =
+      upon (after Spec.tick_interval) (fun () ->
+          Gauge.set gauge (Spec.display !total) ;
+          tick () )
+    in
+    tick ()
+
+  let update datum =
+    total := !total +. datum ;
+    upon (after Spec.rolling_interval) (fun () -> total := !total -. datum)
```

Comment:
Seems like create a new async job whenever we call `update`. Jaiwei said it doesn't seem like we won't call it too much. Are you fine with this @nholland94?

Reply:
I see nothing wrong here. It's not like `after` sits as a job on the async scheduler, eating up time.